common/predictive-text/worker/correction/distance-modeler.ts

```diff
@@ -276,14 +286,15 @@ namespace correction {
     // Signals that the edge has already been processed.
     private processedEdgeSet: {[mapKey: string]: boolean} = {};
 
-    constructor(traversalRoot: LexiconTraversal) {
+    constructor(model: LexicalModel) {
+      if(!model || !model.traverseFromRoot) {
```

Comment:
```suggestion
      if(!model.traverseFromRoot) {
```

Reply:
Maybe it's because of a simplification I put in for unit testing, but we actually need a `null` check on `model` itself _as well as_ on `model.traverseFromRoot`.

Reply:
Haha, if only we could [enable strict null-checking](https://www.typescriptlang.org/docs/handbook/2/basic-types.html#strictnullchecks) in this project! My comments like these are the result of seeing that a lot of these runtime checks could be enforced at compile-time rather than at runtime.